crates/runtime/src/accelerated_table/refresh.rs

```diff
@@ -119,32 +120,37 @@ impl Refresher {
             let future_result = stream.next().await;
 
             match future_result {
-                Some(data_update) => {
-                    let Ok(data_update) = data_update else {
+                Some(result) => {
+                    let Ok((start_time, data_update)) = result else {
                         continue;
                     };
 
-                    if data_update.data.is_empty() {
+                    if data_update.data.is_empty()
+                        || data_update
+                            .data
+                            .first()
+                            .map_or(false, |x| x.columns().is_empty())
+                    {
+                        if let Some(start_time) = start_time {
+                            if let Ok(elapse) = SystemTime::now()
+                                .duration_since(start_time)
+                                .map(|x| x.as_secs())
```

Comment:
Does Rust not have a good time formatter like Go where it picks the best granularity?

Reply:
used a humantime (which is already part of the crate dependencies)